### PR TITLE
fix(avatar): passes color contrast after adjusting text color

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #6203

## Summary

The color of the name initials for light and dark modes are not far apart enough to create interesting background colors that would pass contrast. 

To make light letters lighter and dark letters darker, we’re swapping `ui-text-3` with `ui-text-2`. 
Adding a visual test as well.

### CODE REVIEW

**Review by driskull — APPROVED** [review]
👍

**Review by paulcpederson — APPROVED** [review]

File: src/components/avatar/avatar.stories.ts [hunk 1]
```diff
@@ -35,16 +34,52 @@ export const missingThumbnail = (): string => html`
...
-    dir="rtl"
-    class="calcite-mode-dark"
-    scale="${select("scale", ["s", "m", "l"], "m")}"
-    full-name="${text("full-name", "John Doe")}"
-    username="${text("username", "jdoe")}"
-    user-id="${text("user-id", "9a7c50e6b3ce4b859f7b31e302437164")}"
-  >
-  </calcite-avatar>
-`;
+const users = [
```
Comment (paulcpederson) [inline src/components/avatar/avatar.stories.ts:37]: nice!

File: src/components/avatar/avatar.stories.ts [hunk 2]
```diff
@@ -35,16 +34,52 @@ export const missingThumbnail = (): string => html`
...
+  { id: "8542b186e5a64a90910486de32bced72", name: "qk" },
+  { id: "8587853c1f544de6ae4133224db29736", name: "rl" },
+  { id: "9de33a713bd84690bbc618e98ecf7567", name: "sm" },
+  { id: "e85f7f72aa51b34f81660a0f4c6a4d80", name: "tn" }
+];
+
+const contrastLight = users
+  .map(
+    (user): string =>
+      html`<calcite-avatar key=${user.id} user-id="${user.id}" username="${user.name}"></calcite-avatar>`
```
Comment (jcfranco) [inline src/components/avatar/avatar.stories.ts:63]: `key` shouldn't be needed here as this isn't VDOM. This is generating HTML strings.

File: src/components/avatar/avatar.stories.ts [hunk 3]
```diff
@@ -35,16 +34,52 @@ export const missingThumbnail = (): string => html`
...
+  .join("");
 
-darkModeRTL_TestOnly.parameters = { modes: modesDarkDefault };
+const contrastDarkRTL = users
+  .map(
+    (user): string =>
+      html`<calcite-avatar
+        dir="rtl"
+        class="calcite-mode-dark"
+        key=${user.id}
```
Comment (jcfranco) [inline src/components/avatar/avatar.stories.ts:73]: Same comment as above.

File: src/components/avatar/avatar.stories.ts [hunk 4]
```diff
@@ -35,16 +34,52 @@ export const missingThumbnail = (): string => html`
...
+        user-id="${user.id}"
+        username="${user.name}"
+      ></calcite-avatar>`
+  )
+  .join("");
+
+export const contrast = (): string => html`
+  <div style="display: flex; width: 35%; justify-content: space-between; align-items: center; padding: 25px 0;">
```
Comment (jcfranco) [inline src/components/avatar/avatar.stories.ts:81]: Nitpick: you could introduce a style tag w/ classes to make some CSS reusable.

**Review by jcfranco — APPROVED** [review]
Awesome! 🎉

**Comment by jcfranco** [comment]
Sidebar: I believe our a11y helper for E2E tests check for contrast too. It'd be good to enhance (separately) the existing a11y tests with something similar to what you added in the stories.

**Comment by jcfranco** [comment]
Ignoring UI test change as it's likely caused by a timing issue in the test (screenshot taken before visual test was laid out properly).

**Pull request merged**